e2e/cypress/integration/channel/channel_routing_spec.js

```diff
@@ -0,0 +1,76 @@
+                cy.get('#channelHeaderTitle').should('be.visible').should('contain', 'Private channel');
+                cy.apiDeleteChannel(response.body.id);
+            });
+        });
+    });
+
+    it('should go to self direct channel using the multiple ways to go', () => {
+        cy.apiGetUsers(['user-1']).then((userResponse) => {
+            const user = userResponse.body[0];
+            cy.apiCreateDirectChannel([user.id, user.id]).then((response) => {
+                cy.visit(`/ad-1/channels/${user.id}__${user.id}`);
+                cy.get('#channelHeaderTitle').should('be.visible').should('contain', 'user-1');
+                cy.visit(`/ad-1/channels/${response.body.id}`);
+                cy.get('#channelHeaderTitle').should('be.visible').should('contain', 'user-1');
+                cy.visit(`/ad-1/messages/@${user.username}`);
+                cy.get('#channelHeaderTitle').should('be.visible').should('contain', 'user-1');
+                cy.visit(`/ad-1/messages/${user.email}`);
+                cy.get('#channelHeaderTitle').should('be.visible').should('contain', 'user-1');
+            });
+        });
+    });
+
+    it('should go to other user direct channel using multiple ways to go', () => {
+        cy.apiGetUsers(['user-1', 'sysadmin']).then((userResponse) => {
+            const user1 = userResponse.body[1];
+            const user2 = userResponse.body[0];
+            const userIds = [user2.id, user1.id];
+            cy.apiCreateDirectChannel(userIds).then((response) => {
+                cy.visit(`/ad-1/channels/${user1.id}__${user2.id}`);
+                cy.get('#channelHeaderTitle').should('be.visible').should('contain', 'sysadmin');
+                cy.visit(`/ad-1/channels/${response.body.id}`);
+                cy.get('#channelHeaderTitle').should('be.visible').should('contain', 'sysadmin');
+                cy.visit(`/ad-1/messages/@${user2.username}`);
+                cy.get('#channelHeaderTitle').should('be.visible').should('contain', 'sysadmin');
+                cy.visit(`/ad-1/messages/${user2.email}`);
+                cy.get('#channelHeaderTitle').should('be.visible').should('contain', 'sysadmin');
+            });
+        });
+    });
+    it('should go group channel using group id', () => {
+        const users = ['user-1', 'aaron.peterson', 'sysadmin'];
+        cy.apiGetUsers(users).then((userResponse) => {
+            const userGroupIds = [userResponse.body[2].id, userResponse.body[1].id, userResponse.body[0].id];
+            cy.apiCreateGroupChannel(userGroupIds).then((response) => {
+                cy.visit(`/ad-1/channels/${response.body.name}`);
+                cy.get('#channelHeaderTitle').should('be.visible').should('contain', 'aaron.peterson, sysadmin');
+                cy.visit(`/ad-1/messages/${response.body.name}`);
+                cy.get('#channelHeaderTitle').should('be.visible').should('contain', 'aaron.peterson, sysadmin');
+            });
+        });
+    });
+});
```

Comment:
should we have the case that was failing? this is testing the happy path, but not the unexpected ones like the one mentioned in the original ticket

Reply:
Those checks are done in the unit tests so no need to do it in the E2E tests. I want to test that everything works as it should, following the test pyramid. These tests are very expensive so we should minimize the number of them we have.